packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ImageCaptureHostApiImpl.java

```diff
@@ -0,0 +1,137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import android.content.Context;
+import android.util.Size;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.core.ImageCapture;
+import androidx.camera.core.ImageCaptureException;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.ImageCaptureHostApi;
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+
+public class ImageCaptureHostApiImpl implements ImageCaptureHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+
+  private Context context;
+  private SystemServicesFlutterApiImpl systemServicesFlutterApiImpl;
+
+  @VisibleForTesting public CameraXProxy cameraXProxy = new CameraXProxy();
+
+  public ImageCaptureHostApiImpl(
+      @NonNull BinaryMessenger binaryMessenger,
+      @NonNull InstanceManager instanceManager,
+      @NonNull Context context) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    this.context = context;
+  }
+
+  /**
+   * Sets the context that the {@link ImageCapture} will use to find a location to save a captured
+   * image.
+   */
+  public void setContext(Context context) {
+    this.context = context;
+  }
+
+  /**
+   * Creates a {@link ImageCapture} with the requested flash mode and target resolution if
+   * specified.
+   */
+  @Override
+  public void create(
+      @NonNull Long identifier,
+      @Nullable Long flashMode,
+      @Nullable GeneratedCameraXLibrary.ResolutionInfo targetResolution) {
+    ImageCapture.Builder imageCaptureBuilder = cameraXProxy.createImageCaptureBuilder();
+    if (flashMode != null) {
+      // This sets the requested flash mode, but may fail silently.
+      imageCaptureBuilder.setFlashMode(flashMode.intValue());
```

Comment:
Should there be an error or warning if a flashMode is passed that is larger than an int? Is there a more typesafe variable we can use here like an enum?

Reply:
https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_ON()
The values are not large enough to be a long and in your test you are converting from an int to a long. Why use longs at all?

Reply:
Longs are used in the code generated by pigeon. That's the only reason I'm using them here